### ISSUE TITLE
GDTTransformer not closed

### ISSUE THREAD

**Issue by icedice** [firebase/firebase-ios-sdk#6657]
I am opening a new issue as the other one is closed and the issue I see might not be the same one as I'm certain that I have updated my firestore dependencies to newest version.  I have testet on iOS 10, 12 and 14 devices and I only see this on iOS 14. 

The app never crashes, but we cannot release as long as we see this. Is there anyway I can debug why this is happening?

-------

I tried deleting the whole Carthage folder along with the whole carthage cache folder(~/Library/Caches/org.carthage.CarthageKit) and then bootstrap it again. 
I'm still seeing the issue in both simulator and on actual device. 

here is my cartfile.resolved (relevant part):

```
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseABTestingBinary.json" "6.33.0"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseAnalyticsBinary.json" "6.33.0"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseAuthBinary.json" "6.33.0"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseCrashlyticsBinary.json" "6.33.0"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseFirestoreBinary.json" "6.33.0"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseMessagingBinary.json" "6.33.0"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebasePerformanceBinary.json" "6.33.0"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseProtobufBinary.json" "6.33.0"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseRemoteConfigBinary.json" "6.33.0"

```
I tried inspecting the .framework for a version number but the result looks strange:

```
plutil -p /Carthage/Build/iOS/FirebaseFirestore.framework/info.plist                                                                                                                                                                                                                                                                                                  
{
  "CFBundleExecutable" => "FirebaseFirestore"
  "CFBundleIdentifier" => "com.firebase.Firebase-FirebaseFirestore"
  "CFBundleInfoDictionaryVersion" => "6.0"
  "CFBundleName" => "FirebaseFirestore"
  "CFBundlePackageType" => "FMWK"
  "CFBundleVersion" => "1"
  "DTSDKName" => "iphonesimulator11.2"
}
```


Could this in anyway be related to us applying this fix: https://github.com/Carthage/Carthage/issues/3019#issuecomment-699143260 to accommodate for the Carthage / XC12 issue? As Carthage is simply downloading the binaries for firestore my guess is that is should not be related...

_Originally posted by @icedice in https://github.com/firebase/firebase-ios-sdk/issues/6258#issuecomment-702024388_

**Comment by maksymmalyhin** [comment]
It may potentially be related to #6404.

**Comment by maksymmalyhin** [comment]
@icedice Thank you for creating a separate issue with additional details. Sorry you have the issue. 

As we could not reproduce the issue on our end so far, we will most likely need a sample project with steps to reproduce the issue to make progress.

**Comment by icedice** [comment]
Ok, so I used quite some time researching this. And what I found was embarrassing. It turns out that I was never actually adding the "GoogleDataTransport" framework to my project. As soon as I added it the error wend away. 
The reason that I missed this is that the app was working as expected (except from the warning) on all simulators and devices I tested it on. 

I'm now trying to understand how the app would actually run without actually having "GoogleDataTransport". I suspect it was using a version from a shared cache or something like that but I really don't know. If anyone could explain it it would be great. Having solved the issue is nice, but not really understanding it is frustrating.
